YAML/YAML.sublime-syntax

```diff
@@ -757,10 +751,21 @@ contexts:
       captures:
         1: punctuation.separator.key-value.mapping.yaml
       pop: true
-    - match: (?=\?(?=\s|$))  # Empty mapping keys & values are allowed
+    # multi-line keys are indented by at least one more space than `?`
+    - match: ^(?!\1 +\S|\s*$)
+      pop: true
+    # explicit key of any indentation
+    - match: ^ *(?=\?\s)
       pop: true
```

Comment:
What does this match protect us from? Could you add a test for it? As far as I can see,
```
!!set
? a
? b
```
should already be handled by the earlier match.

Reply:
Probably just for double safety out of fear to break something. It actually seems contra-productive.